projects/addon-editor/components/edit-link/edit-link.component.ts

```diff
@@ -36,15 +36,13 @@ export class TuiEditLinkComponent {
     prefix = 'http://';
 
     constructor(
-        @Inject(DOCUMENT) documentRef: Document,
+        @Inject(DOCUMENT)
+        private readonly documentRef: Document,
         @Optional()
         @Inject(TUI_DOCUMENT_OR_SHADOW_ROOT)
-        shadowRootRef: DocumentOrShadowRoot | null,
+        private readonly shadowRootRef: DocumentOrShadowRoot | null,
     ) {
-        const doc = shadowRootRef || documentRef;
-        const selection = doc.getSelection();
-
-        this.url = selection ? this.getHref(selection) : '';
+        this.url = this.makeUrl();
```

Comment:
Please move these two to property declaration, don't forget the order

Reply:
fixed